Formula/elan-init.rb

```diff
@@ -0,0 +1,54 @@
+class ElanInit < Formula
+  desc "Lean Theorem Prover installer and version manager"
+  homepage "https://leanprover-community.github.io/"
+  url "https://github.com/leanprover/elan/archive/v1.0.3.tar.gz"
+  sha256 "fb3ddd8915a0694ead0f3a51fdf8b0a5540f983f44eee0e757339244c522b8ee"
+  license "Apache-2.0"
+  head "https://github.com/leanprover/elan.git"
+
+  depends_on "rust" => :build
+
+  # elan-init will run on arm64 Macs, but will fetch Leans that are x86_64.
+  depends_on arch: :x86_64
+
+  def install
+    cargo_home = buildpath/"cargo_home"
+    cargo_home.mkpath
+    ENV["CARGO_HOME"] = cargo_home
+    ENV["RELEASE_TARGET_NAME"] = "homebrew-build"
+
+    system "cargo", "install", "--features", "no-self-update", *std_cargo_args
+
+    mv bin/"elan-init", bin/"elan"
+    %w[lean leanpkg leanchecker leanc leanmake].each do |link|
+      ln_s bin/"elan", bin/link
+    end
+
+    system "#{bin}/elan completions bash > elan.bash"
+    system "#{bin}/elan completions zsh > _elan"
+    system "#{bin}/elan completions fish > elan.fish"
+
+    bash_completion.install "elan.bash"
+    zsh_completion.install "_elan"
+    fish_completion.install "elan.fish"
```

Comment:
Can we follow the way other formulae install completions like this? See, for example, `deno`.

Reply:
I copied this from other formulas actually (in this case [diesel](https://github.com/Homebrew/homebrew-core/blob/master/Formula/diesel.rb))` but happy to do it the way `deno` does it instead.

Reply:
`diesel` should probably follow `deno` too 🙂